### PR TITLE
Collisional ionization

### PR DESCRIPTION
**Pull request by chvogl**
This PR will add collisional ionization to the macro atom scheme.

## How Has This Been Tested?
- [x] Testing pipeline
- [ ] Reference Data Comparison following these [instructions](https://tardis-sn.github.io/tardis/development/continuous_integration.html)
- [ ] Other (please describe)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] None of the above (please describe)

## Checklist:

- [ ] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have built the documentation on my fork following [these instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html)
- [x] I have assigned and requested two reviewers for this pull request

### CODE REVIEW

File: tardis/plasma/properties/continuum_processes.py [hunk 1]
```diff
@@ -1014,3 +1016,102 @@ def _calculate_factor(self, nu_i, t_electrons):
...
+    """
+    Attributes
+    ----------
+    p_coll_ion : pandas.DataFrame, dtype float
+        The unnormalized transition probabilities for
+        collisional ionization.
+    p_coll_recomb : pandas.DataFrame, dtype float
+        The unnormalized transition probabilities for
+        collisional recombination.
+    C_coll_ion : pandas.DataFrame, dtype float
```
Comment (andrewfullard) [inline tardis/plasma/properties/continuum_processes.py:1054]: Can this variable have a more descriptive name such as `coll_ion_cool_rate` or similar?

File: tardis/plasma/properties/continuum_processes.py [hunk 2]
```diff
@@ -1014,3 +1016,102 @@ def _calculate_factor(self, nu_i, t_electrons):
...
+    def calculate(
+        self,
+        coll_ion_coeff,
+        coll_recomb_coeff,
+        nu_i,
+        photo_ion_idx,
+        electron_densities,
+        energy_i,
+        level_number_density,
+    ):
```
Comment (andrewfullard) [inline tardis/plasma/properties/continuum_processes.py:1079]: Missing docstring for this method

File: tardis/plasma/properties/continuum_processes.py [hunk 3]
```diff
@@ -1014,3 +1016,102 @@ def _calculate_factor(self, nu_i, t_electrons):
...
+        p_coll_ion = coll_ion_coeff.multiply(energy_i, axis=0)
+        p_coll_ion = p_coll_ion.multiply(electron_densities, axis=1)
+        p_coll_ion = self.set_index(p_coll_ion, photo_ion_idx, reverse=False)
+
+        coll_recomb_rate = coll_recomb_coeff.multiply(
+            electron_densities, axis=1
+        )  # The full rate is obtained from this by multiplying by the
+        # electron density and ion number density.
+        p_recomb_deac = coll_recomb_rate.multiply(nu_i, axis=0) * H
```
Comment (andrewfullard) [inline tardis/plasma/properties/continuum_processes.py:1088]: `deac` is a bit vague. Intended to be deactivate?

File: tardis/plasma/properties/continuum_processes.py [hunk 4]
```diff
@@ -1014,3 +1016,102 @@ def _calculate_factor(self, nu_i, t_electrons):
...
+        p_recomb_deac = self.set_index(
+            p_recomb_deac, photo_ion_idx, transition_type=-1
+        )
+        p_recomb_deac = p_recomb_deac.groupby(level=[0]).sum()
+        index_dd = pd.MultiIndex.from_product(
```
Comment (andrewfullard) [inline tardis/plasma/properties/continuum_processes.py:1093]: Comment describing what this is doing would be nice

File: tardis/plasma/properties/continuum_processes.py [hunk 5]
```diff
@@ -1014,3 +1016,102 @@ def _calculate_factor(self, nu_i, t_electrons):
...
+        p_recomb_internal = coll_recomb_rate.multiply(energy_i, axis=0)
+        p_recomb_internal = self.set_index(
+            p_recomb_internal, photo_ion_idx, transition_type=0
+        )
+        p_coll_recomb = pd.concat([p_recomb_deac, p_recomb_internal])
+
+        C_coll_ion = (coll_ion_coeff * electron_densities).multiply(
+            nu_i * H, axis=0
+        )
+        ll_index = coll_ion_coeff.index
```
Comment (andrewfullard) [inline tardis/plasma/properties/continuum_processes.py:1108]: `ll_index` is another unclear variable name

**Review by andrewfullard — APPROVED** [review]

File: tardis/plasma/properties/continuum_processes.py [hunk 6]
```diff
@@ -86,10 +88,10 @@ def integrate_array_by_blocks(f, x, block_references):
     return integrated
 
 
-# It is currently not possible to use scipy.integrate.cumtrapz in numba.
-# So here is my own implementation.
+# It is currently not possible to use scipy.integrate.cumulative_trapezoid in
+# numba.So here is my own implementation.
```
Comment (chvogl) [inline tardis/plasma/properties/continuum_processes.py:92]:
```suggestion
# numba. So here is my own implementation.
```

**Pull request merged**